core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala

```diff
@@ -646,6 +646,128 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
+      Seq(
+        (UUID.fromString(testData.submission1.submissionId), Seq(testData.indiv1.toReference)),
+        (UUID.fromString(testData.submissionMultipleEntities.submissionId),
+         Vector(testData.indiv1.toReference, testData.indiv2.toReference)
+        )
+      ),
+      runAndWait(
+        submissionEntityQuery.getEntitiesForSubmissions(
+          Seq(UUID.fromString(testData.submission1.submissionId),
+              UUID.fromString(testData.submissionMultipleEntities.submissionId)
+          )
+        )
+      )
+    )
+
+  }
+
+  it should "save entities for a submission" in withDefaultTestDatabase {
+    // Use a submission that has a single entity associated with it
+    // TODO will this mess up other tests
+    assertResult(Seq(testData.indiv1.toReference)) {
+      runAndWait(submissionEntityQuery.getEntitiesForSubmission(UUID.fromString(testData.submission1.submissionId)))
+    }
+
+    // Get the entity ids for some other entities
+    val entity1 = runAndWait(
+      entityQuery
+        .findEntityByName(UUID.fromString(testData.workspace.workspaceId),
+                          testData.sample1.entityType,
+                          testData.sample1.name
+        )
+        .result
+    ).head
+    val entity2 = runAndWait(
+      entityQuery
+        .findEntityByName(UUID.fromString(testData.workspace.workspaceId),
+                          testData.sample2.entityType,
+                          testData.sample2.name
+        )
+        .result
+    ).head
+    // Insert them to the submission
+    runAndWait(
+      submissionEntityQuery.saveEntitiesForSubmission(UUID.fromString(testData.submission1.submissionId),
+                                                      Seq(entity1.id, entity2.id)
+      )
+    )
+
+    assertResult(Set(testData.indiv1.toReference, testData.sample1.toReference, testData.sample2.toReference)) {
+      runAndWait(
+        submissionEntityQuery.getEntitiesForSubmission(UUID.fromString(testData.submission1.submissionId))
+      ).toSet
+    }
+
+  }
+
+  it should "delete the records for a submission" in withDefaultTestDatabase {
+    // These submissions are already saved in the database by the test setup
+
+    // A submission that submitted a list of entityNames
+    assertResult(Seq(testData.indiv1.toReference, testData.indiv2.toReference)) {
+      runAndWait(
+        submissionEntityQuery.getEntitiesForSubmission(
+          UUID.fromString(testData.submissionMultipleEntities.submissionId)
+        )
+      )
+    }
+
+    runAndWait(
+      submissionEntityQuery.deleteRecordsForSubmission(
+        UUID.fromString(testData.submissionMultipleEntities.submissionId)
+      )
+    )
+
+    assertResult(Seq.empty) {
+      runAndWait(
+        submissionEntityQuery.getEntitiesForSubmission(
+          UUID.fromString(testData.submissionMultipleEntities.submissionId)
+        )
+      )
+    }
+  }
+
+  "listWithSubmitter" should "list a submission" in withDefaultTestDatabase {
```

Comment:
`listWithSubmitter` is used from the `listSubmissions` API, but is not tested here; it's incidentally tested at higher levels such as `SubmissionsSpec`.